### PR TITLE
Add currency context for new admin API

### LINKED ISSUE #36483: Missing currency context for new Admin API
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Unable to use currency context with new API with API Platform
image

### Expected behavior

_No response_

### Steps to reproduce

Create a new API ressource who need the currency context in its CQRSQuery, e.g.: GetOrderForViewing and call it.

### PrestaShop version(s) where the bug happened

9.0.0 alpha 1

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by clement-hvt**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add currency context for new admin API
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Create a new API ressource who need the currency context in its CQRSQuery, e.g.: GetOrderForViewing and call the endpoint.
| UI Tests          | https://github.com/clement-hvt/ga.tests.ui.pr/actions/runs/9766900904
| Fixed issue or discussion?     | Fixes #36483
| Related PRs       |
| Sponsor company   |

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

Please consider opening an issue before submitting a Pull Request:
* If it's a bug fix, it helps maintainers verify that the bug is effectively due to a defect in the code, and that it hasn't been fixed already.
* It can help trigger a discussion about the best implementation path before a single line of code is written.
* It may lead the Core Product team to mark that issue as a priority, further attracting the maintainers' attention.

(Note: this is an automated message, but answering it will reach a real human)

**Comment by ps-jarvis** [comment]
Hello @clement-hvt!

This is your first pull request on [PrestaShop](https://github.com/PrestaShop/PrestaShop/) repository of the PrestaShop project.

Thank you, and welcome to this Open Source community!

**Review by tleon — CHANGES_REQUESTED** [review]
Hello, 
Could you add a unit test for this new context ? 
The other liseners' tests can be a good example ( e.g Tests/Unit/PrestaShopBundle/EventListener/Admin/Context/LanguageContextListenerTest.php ) .

Thx for the contribution.

**Comment by clement-hvt** [comment]
> Hello, Could you add a unit test for this new context ? The other liseners' tests can be a good example ( e.g Tests/Unit/PrestaShopBundle/EventListener/Admin/Context/LanguageContextListenerTest.php ) .
> 
> Thx for the contribution.

Unit test added :white_check_mark:

**Review by tleon — DISMISSED** [review]
you need to fix the ci but it's ok for me when it's done

**Review by boherm — DISMISSED** [review]
Thanks for your contribution @clement-hvt!
LGTM too, it's ok after fixing the CI 🙂

**Review by boherm — DISMISSED** [review]

**Review by tleon — DISMISSED** [review]

**Comment by florine2623** [comment]
@tleon ,
Could this be QA by dev ?
It hasn't been Verified on the issue yet ^^

**Comment by tleon** [comment]
Yes this should be QA by dev since you need to create a new test endoint unsing the new context.
I don't know maybe @clement-hvt will propose a PR on the ps_apiressource module that implement such context. If so we could be using this PR as a test.

**Comment by kpodemski** [comment]
@tleon he provided example in the issue, you can use it

**Comment by boherm** [comment]
Hi @clement-hvt! I've just test your PR, and it's ok! :)

**Comment by boherm** [comment]
Can you please rebase to develop and restart UI tests?
Then, we could merge!

**Comment by clement-hvt** [comment]
@boherm [https://github.com/clement-hvt/ga.tests.ui.pr/actions/runs/11139148202](https://github.com/clement-hvt/ga.tests.ui.pr/actions/runs/11139148202)

**Comment by nicosomb** [comment]
@clement-hvt you added a merge commit in your PR and our CI is not happy with that 🙂

**Review by boherm — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
